model/activist.go

```diff
@@ -1155,7 +1172,23 @@ func UpdateActivistData(db *sqlx.DB, activist ActivistExtra, userEmail string) (
 				fmt.Println("ERROR updating activist on mailing list:", err.Error())
 			}
 		}
-		fmt.Println("Updated: ", activist.Location, activist.City, activist.State)
+		if activistAddressChanged && activist.StreetAddress != "" && activist.City != "" && activist.State != "" {
+			full_address := url.QueryEscape(activist.StreetAddress + " " + activist.City + " " + activist.State)
+			request := "https://maps.googleapis.com/maps/api/geocode/json?address=" + full_address + "&key=" + config.GooglePlacesAPIKey
+			resp, err := http.Get(request)
+			if err != nil {
+				fmt.Println("Error geocoding activist location", err)
```

Comment:
if there is an error might be better to break out of this `if` somehow (or return early if you split this into another function)

Reply:
On this note, I think we'd probably prefer to silently fail (& log it) but still go through w/ saving the activist in the db if the gecocoding service fails for w/e reason.

I feel it's pretty important that activist data is still able to be updated even if the geocoding service is failing.

Reply:
Thinking about this further... maybe it would be ideal to just fire off a goroutine here that handles the things that need to happen in the background, such as updating mailing list and coords? B/c we don't really want the user to have to wait around for those things to happen before the activist is saved in the db. (This was a bit of a flaw in the original logic for updating the mailing list.)

Reply:
probably the coords should still be looked synchronously because the result is used in the database update? if so, could the signup be made async in a separate PR so Jeff's change isn't held up?

just wondering, is the async idea to make it faster when an admin of ADB clicks save to update a user? did you notice the adb signup was slow?

maybe whatever is simplest/easiest is best for this probably rarely used feature. but some possible improvements mainly as thought experiments:
* it would be nice to know if the signup operations fails, maybe by an email notification to our team, perhaps directly or by monitoring error logs.
* it would be nice to ensure async operations complete even if the web server container is requested to shut down or even if an error in the main function causes it to return and shut down. perhaps there would be a work queue that the signup gets added to, and the main function can wait for the queue on the last line before exiting.

Reply:
Fair points. Yeah, I was thinking the main reason would be so it saves faster, but I that might be a bit of an over optimization for now. I def won't block on it. 

We have log drains in AWS and can set up Cloudwatch alerts for whatever.

The signup service has its own queue. That being said, I do think the background jobs that run in ADB could use a huge overhaul at some point if we have bandwidth. Right now they only run within a single instance of the app (via an environment variable that enables them). It would be great to implement a more robust job system.

Reply:
Maybe walking through how the signup service works at the next meeting would be useful. A lot of stuff feeds in and out of it, and would be good for others to know how it works.